### ISSUE TITLE
separate rule violations with the same line, column and rule name cannot both create a TODO

### ISSUE THREAD

**Issue by runspired** [ember-template-lint/ember-template-lint#2333]
Example failure (1 TODO created, 1 violation not converted to TODO)

```
  2:2  todo  <div> has a role of tab, it cannot have semantic descendants like <h5>  require-presentational-children
  2:2  error  <div> has a role of tab, it cannot have semantic descendants like <a>  require-presentational-children
  ```
  
  note the offending `h5` and `a` elements are not on line 2 in the associated file, that is merely where the div opens.
  
  Offending content (sorry for the 😱 code)
  
  ```hbs
  <div class="card analytics-runner-card no-radius">
    <div
    class="card-header analytics-runner-card-header no-radius"
    role="tab"
    id="columnsAria"
  >
    <h5 class="mb-0">
      <a
        data-toggle="collapse"
        class="nav-link"
        data-parent="#accordion"
        href="#columns"
        aria-expanded="false"
        aria-controls="columns"
      >
        {{t "analytics.column_order"}}
      </a>
    </h5>
  </div>
 </div>
  ```

**Comment by bryanhickerson** [comment]
I have the same issue, I had to turn off `require-presentational-children` as a result.

**Comment by rwjblue** [comment]
Hmmm. Seems to me like the issue is with which node we are reporting as incorrect (RE: line + col). We should either report only a single error for this case (which would make todo's work fine), or report individual errors but using the specific invalid child as the line+col instead of always using the parent's node.

**Comment by drewlee** [comment]
I've run into the same issue but using `ember-template-lint` v4.3.0 and single file todos. Given the following example markup will yield 2 lint violations.
```html
<div role="button">
  <p>Text content</p>
  <strong>Supplementary content</strong>
</div>
```
```shell
  7:0  error  <div> has a role of button, it cannot have semantic descendants like <p>  require-presentational-children
  7:0  error  <div> has a role of button, it cannot have semantic descendants like <strong>  require-presentational-children

✖ 2 problems (2 errors, 0 warnings)
error Command failed with exit code 1.
```
After generating todos, running `ember-template-lint` results in the following output.
```shell
$ ember-template-lint .
app/templates/application.hbs
  7:0  error  <div> has a role of button, it cannot have semantic descendants like <strong>  require-presentational-children

✖ 1 problems (1 errors, 0 warnings)
error Command failed with exit code 1.
```

Even though todos were generated for both violations, the todos parser doesn't seem to associate the second todo with the error because both todos have the same exact fingerprint.
https://github.com/drewlee/etl-todos-pres-children-bug/blob/master/.lint-todo

I created a simple reproduction case at https://github.com/drewlee/etl-todos-pres-children-bug.

CC @scalvert

**Comment by scalvert** [comment]
This does seem like we're incorrectly reporting the line/column for the violation. If the same rule is flagged with the same line/column, it's considered the same. 

I suspect this is a bug in the rule, not todos.

**Comment by scalvert** [comment]
I suspect [this loop](https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/require-presentational-children.js#L97) should be using `child`, not `node`.
